### PR TITLE
Refactor AIR

### PR DESCRIPTION
**Pull request by ColoCarletti**
# Refactor AIR

This PR resumes the changes made for AIR in the following PRs: [834](https://github.com/lambdaclass/lambdaworks/pull/834) and [697](https://github.com/lambdaclass/lambdaworks/pull/697).

Description of the pull request changes and motivation.

## Type of change

Please delete options that are not relevant.

- [X] Refactor

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [ ] This change is an Optimization
  - [ ] Benchmarks added/run

### CODE REVIEW

File: provers/stark/src/prover.rs [hunk 1]
```diff
@@ -170,17 +171,35 @@ where
...
     /// Returns the Merkle tree and the commitment to the vectors `vectors`.
-    fn batch_commit<E>(vectors: &[Vec<FieldElement<E>>]) -> (BatchedMerkleTree<E>, Commitment)
+    fn batch_commit_main(
+        vectors: &[Vec<FieldElement<A::Field>>],
+    ) -> Option<(BatchedMerkleTree<A::Field>, Commitment)>
+    where
+        FieldElement<A::Field>: AsBytes + Sync + Send,
+        // FieldElement<A::FieldExtension>: AsBytes + Sync + Send,
+        // E: IsSubFieldOf<A::FieldExtension>,
+        // A::Field: IsSubFieldOf<E>,
```
Comment (entropidelic) [inline provers/stark/src/prover.rs:181]: this can be removed

File: provers/stark/src/prover.rs [hunk 2]
```diff
@@ -170,17 +171,35 @@ where
...
+        vectors: &[Vec<FieldElement<A::FieldExtension>>],
+    ) -> Option<(BatchedMerkleTree<A::FieldExtension>, Commitment)>
     where
         FieldElement<A::Field>: AsBytes + Sync + Send,
         FieldElement<A::FieldExtension>: AsBytes + Sync + Send,
-        FieldElement<E>: AsBytes + Sync + Send,
-        E: IsSubFieldOf<A::FieldExtension>,
-        A::Field: IsSubFieldOf<E>,
+        // E: IsSubFieldOf<A::FieldExtension>,
+        // A::Field: IsSubFieldOf<E>,
```
Comment (entropidelic) [inline provers/stark/src/prover.rs:197]: ditto

File: provers/stark/src/prover.rs [hunk 3]
```diff
@@ -191,25 +210,81 @@ pub trait IsStarkProver<A: AIR> {
...
+        }
+
+        // Compute commitment.
+        let lde_trace_permuted_rows = columns2rows(lde_trace_permuted);
+
+        let (lde_trace_merkle_tree, lde_trace_merkle_root) =
+            Self::batch_commit_main(&lde_trace_permuted_rows)?;
+
+        // let (lde_trace_merkle_tree, lde_trace_merkle_root) =
+        //     Self::batch_commit_main(&lde_trace_permuted_rows);
```
Comment (entropidelic) [inline provers/stark/src/prover.rs:248]: ditto

File: provers/stark/src/prover.rs [hunk 4]
```diff
@@ -191,25 +210,81 @@ pub trait IsStarkProver<A: AIR> {
...
+        transcript: &mut impl IsTranscript<A::FieldExtension>,
+    ) -> Option<(
+        Vec<Polynomial<FieldElement<A::FieldExtension>>>,
+        Vec<Vec<FieldElement<A::FieldExtension>>>,
+        BatchedMerkleTree<A::FieldExtension>,
+        Commitment,
+    )>
+    where
+        FieldElement<A::Field>: AsBytes + Send + Sync,
+        // FieldElement<E>: AsBytes + Send + Sync,
```
Comment (entropidelic) [inline provers/stark/src/prover.rs:281]: ditto

File: provers/stark/src/prover.rs [hunk 5]
```diff
@@ -191,25 +210,81 @@ pub trait IsStarkProver<A: AIR> {
...
+        FieldElement<A::FieldExtension>: AsBytes + Send + Sync,
+        // E: IsSubFieldOf<A::FieldExtension> + IsFFTField,
```
Comment (entropidelic) [inline provers/stark/src/prover.rs:283]: ditto

File: provers/stark/src/prover.rs [hunk 6]
```diff
@@ -242,10 +318,11 @@ pub trait IsStarkProver<A: AIR> {
         domain: &Domain<A::Field>,
     ) -> Vec<Vec<FieldElement<E>>>
     where
-        FieldElement<A::Field>: Send + Sync,
         FieldElement<E>: Send + Sync,
+        FieldElement<A::Field>: Send + Sync,
         E: IsSubFieldOf<A::FieldExtension>,
         A::Field: IsSubFieldOf<E>,
+        // F: IsFFTField,
```
Comment (entropidelic) [inline provers/stark/src/prover.rs:325]: ditto

File: provers/stark/src/prover.rs [hunk 7]
```diff
@@ -471,32 +561,40 @@ pub trait IsStarkProver<A: AIR> {
...
             core::iter::successors(Some(FieldElement::one()), |x| Some(x * &gamma))
-                .take(n_terms_composition_poly + n_terms_trace)
+                .take(n_terms_composition_poly + num_terms_trace)
                 .collect();
 
-        let trace_poly_coeffients: Vec<_> = deep_composition_coefficients
-            .drain(..n_terms_trace)
+        // let trace_poly_coeffients: Vec<_> = deep_composition_coefficients
+        //     .drain(..num_terms_trace)
+        //     .collect();
```
Comment (entropidelic) [inline provers/stark/src/prover.rs:577]: ditto

File: provers/stark/src/table.rs [hunk 8]
```diff
@@ -73,12 +73,12 @@ impl<'t, F: IsField> Table<F> {
...
-        let n_cols = self.width;
-        let row_offset = row_idx * n_cols;
-        &mut self.data[row_offset..row_offset + n_cols]
-    }
+    // /// Given a row index, returns a mutable reference to that row as a slice of field elements.
+    // pub fn get_row_mut(&mut self, row_idx: usize) -> &mut [FieldElement<F>] {
+    //     let n_cols = self.width;
+    //     let row_offset = row_idx * n_cols;
+    //     &mut self.data[row_offset..row_offset + n_cols]
+    // }
```
Comment (entropidelic) [inline provers/stark/src/table.rs:81]: ditto

File: provers/stark/src/trace.rs [hunk 9]
```diff
@@ -347,20 +386,20 @@ pub fn columns2rows<F: IsField>(columns: Vec<Vec<FieldElement<F>>>) -> Vec<Vec<F
...
-        let res_cols = trace_table.columns();
+//         let trace_table = TraceTable::from_columns(vec![col_1.clone(), col_2.clone()], 2, 1);
+//         let res_cols = trace_table.columns();
 
-        assert_eq!(res_cols, vec![col_1, col_2]);
-    }
-}
+//         assert_eq!(res_cols, vec![col_1, col_2]);
+//     }
+// }
```
Comment (entropidelic) [inline provers/stark/src/trace.rs:405]: ditto

**Review by entropidelic — APPROVED** [review]
lgtm, just remove the commented unused code :)

**Review by MauroToscano — APPROVED** [review]

**Pull request merged**